js/client/modules/@arangodb/arango-collection.js

```diff
@@ -535,22 +535,21 @@ ArangoCollection.prototype.truncate = function (options) {
   } else {
     options = options || {};
   }
-
+  if (!options.hasOwnProperty('compact')) {
+    options.compact = true;
+  }
   let headers = {};
   if (options && options.transactionId) {
     headers['x-arango-trx-id'] = options.transactionId;
   }
 
   var append = (options.waitForSync ? '&waitForSync=true' : '');
+  append += (append === '')? '?':'&' + (options.compact ? 'compact=true' : 'compact=false');
```

Comment:
It looks like the generated URL will have a wrong format if only `waitForSync` is specified.
This is unrelated to this PR, but we should fix it as well:
```suggestion
  var append = (options.waitForSync ? '?waitForSync=true' : '');
  append += (append === '') ? '?' : '&' + (options.compact ? 'compact=true' : 'compact=false');
```